lib/services/local/presence.go

```diff
@@ -678,64 +678,66 @@ func (s *PresenceService) DeleteAllTunnelConnections() error {
 }
 
 // CreateRemoteCluster creates remote cluster
-func (s *PresenceService) CreateRemoteCluster(rc types.RemoteCluster) error {
+func (s *PresenceService) CreateRemoteCluster(
+	ctx context.Context, rc types.RemoteCluster,
+) (types.RemoteCluster, error) {
 	value, err := json.Marshal(rc)
 	if err != nil {
-		return trace.Wrap(err)
+		return nil, trace.Wrap(err)
 	}
 	item := backend.Item{
 		Key:     backend.Key(remoteClustersPrefix, rc.GetName()),
 		Value:   value,
 		Expires: rc.Expiry(),
 	}
-	_, err = s.Create(context.TODO(), item)
+	lease, err := s.Create(ctx, item)
 	if err != nil {
-		return trace.Wrap(err)
+		return nil, trace.Wrap(err)
 	}
-	return nil
+	rc.SetRevision(lease.Revision)
+	return rc, nil
 }
 
 // UpdateRemoteCluster updates selected remote cluster fields: expiry and labels
 // other changed fields will be ignored by the method
-func (s *PresenceService) UpdateRemoteCluster(ctx context.Context, rc types.RemoteCluster) error {
+func (s *PresenceService) UpdateRemoteCluster(ctx context.Context, rc types.RemoteCluster) (types.RemoteCluster, error) {
```

Comment:
Should we use the revision here from the passed in RC or the one that's fetched and updated ? Maybe I need to add a manual revision comparison between `rc` and `existing` here.